src/lib/FreeC/Backend/Agda/Syntax.hs

```diff
@@ -153,10 +200,36 @@ ident = Ident . qname' . name
 
 -- | Hides the given expression.
 --
---   e ↦ {e}
+--   > e ↦ {e}
 hiddenArg_ :: Expr -> Expr
 hiddenArg_ = HiddenArg NoRange . unnamed
 
+-- | @if_then_else_@ from the base library.
+--
+--   > cond true false ↦ if cond then true else false
+ite :: Expr -> Expr -> Expr -> Expr
+ite cond true false =
+  RawApp NoRange [ident "if", cond, ident "then", true, ident "else", false]
```

Comment:
Can we find a better name for this smart constructor?

Reply:
`ite` was the best name i could think of. `ifThenElse` seemed to long and `ite P Q R` is a common abbreviation for `(P → Q) ∧ (¬P → R)` in propositional logic. Since it's only used once the length of the name doesnt really matter. If you prefer `ifThenElse` or another name I'll change the it.

Reply:
Yeah, I'd prefer `ifThenElse`. That's a good name. It is also not particularly long compared to other function names in the code base.